src/angular-masonry-directive.js

```diff
@@ -1,71 +1,77 @@
 (function() {
     "use strict";
 
-    angular.module('masonry', ['ng']).directive('masonry', function($timeout) {
+    angular.module('masonry', ['ng']).directive('masonry', ['$timeout', function($timeout) {
         return {
-            restrict: 'AC',
+            restrict: 'A',
+            scope: true,
             link: function(scope, elem, attrs) {
                 var container = elem[0];
-                var options = angular.extend({
+                var options = {
                     itemSelector: '.item'
-                }, angular.fromJson(attrs.masonry));
+                }
+                if (attrs.masonry) {
+                    angular.extend(options, angular.fromJson(attrs.masonry));
+                }
 
                 var masonry = scope.masonry = new Masonry(container, options);
 
-                var debounceTimeout = 0;
-                scope.update = function() {
+                var debounceTimeout = null;
+                scope.updateMasonry = function() {
                     if (debounceTimeout) {
                         $timeout.cancel(debounceTimeout);
                     }
                     debounceTimeout = $timeout(function() {
-                        debounceTimeout = 0;
+                        debounceTimeout = null;
 
                         masonry.reloadItems();
                         masonry.layout();
-    
+
                         elem.children(options.itemSelector).css('visibility', 'visible');
-                    }, 120);
```

Comment:
I'm actually not sure why these high timeouts exist (and not 0 to append to the end of the digest queue). Hope you could shed some light on it.